packages/lastrev-adapter-contentful/src/contentfulAdapter/contentfulAdapter.ts

```diff
@@ -17,16 +17,19 @@ const Adapter = <T>({
   contentRefTypeText = 'Content reference',
   assetRefTypeText = 'Asset reference',
   contentUrlLookup,
-  skipContentTypes
+  skipContentTypes,
+  maxTraverseDepth = 10
 }: AdapterConfig): Transform => (data) => {
   const parsedEntries: Record<string, ParsedEntry> = {};
 
-  const traverse = (obj: unknown) => {
+  const traverse = (obj: unknown, maxDepth: number = maxTraverseDepth) => {
+    if (maxDepth < 0) return null;
+
     if (isBadContentfulObject(obj)) {
       return null;
     }
     if (isArray(obj)) {
-      return compact(map(obj, traverse)) as unknown[];
+      return compact(map(obj, (x) => traverse(x, maxDepth - 1))) as unknown[];
```

Comment:
I don't think maxDepth should be decremented here. Only should be decremented at the entry level.